core/src/test/resources/newlines/alwaysBeforeTopLevelStatements.stat

```diff
@@ -34,29 +37,37 @@ package a {
       val v1 = 1
       val v2 = 2
 
-      case class C1
-
-      sealed trait T1
-
-      final class FC1
-
+      val v3 = {
+        3
+      }
       def a = v1
-
       def b = v2
-
       def c: Int
-
       def d: Int = 2
 
+      def e: Int = {
+        3
+      }
+
       def e = a match {
         case 1 => true
         case 2 => false
       }
     }
   }
 }
-
-
+<<< single line
+object a {
+case class C1
+sealed trait T1
+final class FC1
+}
+>>>
+object a {
+  case class C1
```

Comment:
BTW @Daxten , following @dwijnand request, I changed the behavior to not force blank lines on single-line class/object/trait. I think this makes sense for ADTs,

```scala
sealed trait MyEnum
object MyEnum {
  case class A() extends MyEnum
  case object Foo extends MyEnum
}
```